R/baselinenowcast.R

```diff
@@ -152,3 +164,280 @@ baselinenowcast.reporting_triangle <- function(
+#'   assert_date
+#' @family baselinenowcast_df
+#' @export
+#' @method baselinenowcast data.frame
+#' @returns Data.frame of class \code{\link{baselinenowcast_df}}
+#' @examples
+#' covid_data_to_nowcast <- germany_covid19_hosp[
+#'   germany_covid19_hosp$report_date <
+#'     max(germany_covid19_hosp$reference_date),
+#' ] # nolint
+#' nowcasts_df <- baselinenowcast(covid_data_to_nowcast,
+#'   max_delay = 40,
+#'   strata_cols = c("age_group", "location")
+#' )
+#' nowcasts_df
+baselinenowcast.data.frame <- function(
+    data,
+    scale_factor = 3,
+    prop_delay = 0.5,
+    output_type = c("samples", "point"),
+    draws = 1000,
+    uncertainty_model = fit_by_horizon,
+    uncertainty_sampler = sample_nb,
+    max_delay,
+    delays_unit = "days",
+    strata_cols = NULL,
+    strata_sharing = "none",
+    ...) {
+  output_type <- arg_match(output_type)
+  assert_names(colnames(data),
+    must.include = c("reference_date", "report_date", "count")
+  )
+  assert_character(delays_unit)
+  assert_character(strata_sharing)
+  assert_character(output_type)
+  assert_subset(strata_sharing,
+    choices = c("delay", "uncertainty", "none")
+  )
+  if (length(strata_sharing) == 2 && "none" %in% strata_sharing) {
+    cli_abort(
+      message = c("`strata_sharing` cannot be both 'none' and 'delay'/'uncertainty'") # nolint
+    )
+  }
+  # Filter to max delay
+  data$delay <- as.numeric(
+    difftime(
+      as.Date(data$report_date),
+      as.Date(data$reference_date),
+      units = delays_unit
+    )
+  )
+  data_clean <- data[data$delay <= max_delay, ]
+
+  .validate_strata_cols(
+    strata_cols,
+    data_clean
+  )
+  assert_date(data_clean$reference_date)
+  assert_date(data_clean$report_date)
+
+  # Split dataframe into a list of dataframes for each strata
+  list_of_dfs <- .split_df_by_cols(
+    long_df = data_clean,
+    col_names = strata_cols
+  )
+
+  # Create a list of reporting triangles
+  list_of_rep_tris <- lapply(list_of_dfs,
+    as_reporting_triangle,
+    max_delay = max_delay,
+    delays_unit = delays_unit
+  )
+  # Combine if needed
+  shared_delay_pmf <- NULL
+  shared_uncertainty_params <- NULL
+  if (all(strata_sharing != "none")) {
+    pooled_df <- .combine_triangle_dfs(
+      data = data_clean,
+      strata_cols = strata_cols
+    )
+    pooled_triangle <- as_reporting_triangle(pooled_df,
+      max_delay = max_delay
+    )
+    # Get the training volume for all reporting triangles
+    tv <- allocate_reference_times(
+      reporting_triangle = pooled_triangle$reporting_triangle_matrix,
+      scale_factor = scale_factor,
+      prop_delay = prop_delay
+    )
+    if ("delay" %in% strata_sharing) {
+      # Estimate delay once on pooled data
+      shared_delay_pmf <- estimate_delay(
+        reporting_triangle = pooled_triangle$reporting_triangle_matrix,
+        n = tv$n_history_delay
+      )
+    }
+    if ("uncertainty" %in% strata_sharing) {
+      # Estimate uncertainty once on pooled data
```

Comment:
I am not sure why this would mean you need to change the underlying function call? You could be able to pass uncertainty_model here like this even if it is not defined specifically in the function call? If it really is a problem the swapping out to a do.call approach will do the trick

Reply:
Fixed this, my bad